chart/tests/test_security_context.py

```diff
@@ -0,0 +1,206 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import jmespath
+
+from tests.helm_template_generator import render_chart
+
+
+class TestSCBackwardsCompatibility:
+    def test_check_deployments_and_jobs(self):
+        docs = render_chart(
+            values={
+                "uid": 3000,
+                "gid": 30,
+                "webserver": {"defaultUser": {"enabled": True}},
+                "flower": {"enabled": True},
+                "airflowVersion": "2.2.0",
+                "executor": "CeleryKubernetesExecutor",
+            },
+            show_only=[
+                "templates/flower/flower-deployment.yaml",
+                "templates/scheduler/scheduler-deployment.yaml",
+                "templates/triggerer/triggerer-deployment.yaml",
+                "templates/webserver/webserver-deployment.yaml",
+                "templates/workers/worker-deployment.yaml",
+                "templates/jobs/create-user-job.yaml",
+                "templates/jobs/migrate-database-job.yaml",
+            ],
+        )
+
+        for index in range(len(docs)):
+            assert 3000 == jmespath.search("spec.template.spec.securityContext.runAsUser", docs[index])
+            assert 30 == jmespath.search("spec.template.spec.securityContext.fsGroup", docs[index])
+
+    def test_check_statsd_uid(self):
+        docs = render_chart(
+            values={"statsd": {"enabled": True, "uid": 3000}},
+            show_only=["templates/statsd/statsd-deployment.yaml"],
+        )
+
+        for index in range(len(docs)):
+            assert 3000 == jmespath.search("spec.template.spec.securityContext.runAsUser", docs[index])
```

Comment:
```suggestion
        assert 3000 == jmespath.search("spec.template.spec.securityContext.runAsUser", docs[0])
```
Both here and elsewhere.